experimental/vjn/yarpl/main/main.cc

```diff
@@ -0,0 +1,80 @@
+#include <chrono>
+#include <iostream>
+#include <memory>
+#include <thread>
+
+#include <reactivestreams/subscriber.h>
+#include <yarpl/flowable.h>
+#include <yarpl/scheduler.h>
+
+namespace {
+
+using reactivestreams::Subscriber;
+
+auto make_subscriber(int state) {
+  return Subscriber<int>::create(
+      [state](int i) mutable {
+        std::cout << "  state: " << ++state << ", i = " << i << std::endl;
+      },
+      [](const std::exception_ptr eptr) {
+        try {
+          std::rethrow_exception(eptr);
+        } catch (const std::exception& exception) {
+          std::cerr << "  exception: " << exception.what() << std::endl;
+        } catch (...) {
+          std::cerr << "  !unknown exception!" << std::endl;
+        }
+      },
+      []() { std::cout << "  complete." << std::endl; });
+}
+
+class Scheduler : public yarpl::Scheduler {
```

Comment:
can we name the Scheduler such that it represents that for each task it spins a new thread?